### PR TITLE
Add collabora online support

### PR DESCRIPTION
**Pull request by moan0s**
This PR should implement collabora online support as document server. It uses the [official CODE image](https://hub.docker.com/r/collabora/code/). I tested the role successfully with my external nginx. As I never worked with traefik and don't know how to make it work with that I marked this PR as WIP and would ask for support in that regard. Any help is welcome.

Feedback for all other parts of course also very welcome

### CODE REVIEW

**Comment by moan0s** [comment]
For traefik this could help: https://goneuland.de/collabora-office-online-mit-docker-compose-und-traefik-installieren/

```
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.collabora.entrypoints=https"
      - "traefik.http.routers.collabora.rule=(Host(`collabora.euredomain.de`))"
      - "traefik.http.routers.collabora.tls=true"
      - "traefik.http.routers.collabora.tls.certresolver=http"
      - "traefik.http.routers.collabora.service=collabora"
      - "traefik.http.services.collabora.loadbalancer.server.port=9980"
      - "traefik.docker.network=proxy"
      - "traefik.http.routers.collabora.middlewares=default@file" 
```

File: docs/configuring-playbook-collabora.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+The above only does basic configuration, hooking your Collabora document server with Nextcloud.
+The Nextcloud office app supports additional options, which you can configure manually from: **Administration Settings** -> **Office**.
+
+For security reasons you should set the `Allow list for WOPI requests` to match the IP of your server. Read more under [here](https://docs.nextcloud.com/server/latest/admin_manual/office/configuration.html#wopi-settings)
+
+
+## Reverse Proxy Configuration
+
+If you use an external reverse proxy you have to configure it correctly. Here is an example of a working nginx configuration.
```
Comment (spantaleev) [inline docs/configuring-playbook-collabora.md:53]: If you're using nginx, why not use the playbook to manage the configuration for you, so you can just include it into your own nginx webserver?

See the **Using nextcloud-nginx-proxy to generate configuration for your own, other, nginx proxy** section in `docs/configuring-playbook-reverse-proxy.md`

This should make things easier for people using nginx (including you) and should simplify this documentation page.

---

Besides this, we should also add Traefik support, as that's the default for this playbook and a much better way to reverse-proxy to containers, especially when running various web-based services on the same host.

File: docs/configuring-playbook-collabora.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+|-------|-------------|----------|--------|------|----------------------------|
+| CNAME | `collabora` | -        | -      | -    | `nextcloud.yourdomain.org` |
+
+## `vars.yml` configuration
+
+Enable the server like this in your configuration file (`inventory/host_vars/<your-domain>/vars.yml`) and set a password
+for the admin interface.
+
+```yaml
+nextcloud_collabora_document_server_enabled: true
```
[the remark this answers is not in the document]
Comment (moan0s) [reply docs/configuring-playbook-collabora.md:26]: Probably not - yes it was carried over.
We could go for `_collabora_online` or `_code`. I think for clarity I'd prefer `_collabora_online` as code has so many different meanings and we will probably never integrate the non-rolling release version
https://sdk.collaboraonline.com/docs/faq.html#what-is-code-are-there-non-development-editions-too

[the remark this answers is not in the document]
Comment (spantaleev) [reply docs/configuring-playbook-collabora.md:26]: Collabora Online sounds good!

File: roles/custom/nextcloud_collabora_document_server/templates/systemd/nextcloud-collabora-document-server.service.j2 [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+ExecStart={{ devture_systemd_docker_base_host_command_docker }} run \
+			--rm \
+			--name=nextcloud-collabora-document-server \
+			--log-driver=none \
+			-p 127.0.0.1:9980:9980 \
+			-e "aliasgroup1=https://{{ nextcloud_server_fqn_nextcloud |replace(".", "\\\\.") }}:443" \
+			-e "username=admin" \
+			-e "password="{{ nextcloud_collabora_document_server_password }}" \
+			-e "extra_params=--o:ssl.enable=true" \
```
[the remark this answers is not in the document]
Comment (moan0s) [reply roles/custom/nextcloud_collabora_document_server/templates/systemd/nextcloud-collabora-document-server.service.j2]: Yes the lower case variables are used in the documentation 🤷

File: roles/custom/nextcloud_collabora_document_server/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+
+nextcloud_collabora_document_server_enabled: false
```
[the remark this answers is not in the document]
Comment (moan0s) [reply roles/custom/nextcloud_collabora_document_server/defaults/main.yml]: Done

File: docs/configuring-playbook-collabora.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+nextcloud_collabora_document_server_password: 'lHAbqkYapmelxLWFqjrYS3v9RQtIzQbWrvs'
+```
+
+You probably should also adjust the domain name to match the DNS record (the default value is `collabora.{{ nextcloud_server_fqn_nextcloud }}`)
+
+```yaml
+nextcloud_collabora_document_server_domain: "collabora.yourdomain.org"
+```
```
[the remark this answers is not in the document]
Comment (moan0s) [reply docs/configuring-playbook-collabora.md:34]: Done

File: docs/configuring-playbook-collabora.md (hunk 1 above)
Comment (moan0s) [reply docs/configuring-playbook-collabora.md:61]: I gave it a try but I have a hard time to wrap my head around it. If I understand correctly we would aim the default behavior to flow `Traefik`->`nextcloud-collabora-online` and an alternative as `nextcloud-nginx-proxy` -> `collabora-online` and `my reverse proxy` -> `collabora-online` (which is the case for me so I'd use `nextcloud_nginx_proxy_service_reach_mode: local` to use the configurationn generated  by nextcloud-nginx-proxy in my own nginx right?

File: docs/configuring-playbook-collabora.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
+# Setting up Collabora integration
+
+**WARNING**: Work in progress - Currently only external reverse proxy supported.
+
+The following configuration enables you to install the [Collabora Online Development Edition (CODE)](https://www.collaboraoffice.com/) document server.
+You will have a fully functioning [WOIP server]() that (together with the [Nextcloud Office App](https://apps.nextcloud.com/apps/richdocuments)) enables you to
```
Comment (spantaleev) [inline docs/configuring-playbook-collabora.md:6]: WOIP -> WOPI? Where should this link? Maybe [here](https://learn.microsoft.com/en-us/openspecs/office_protocols/ms-wopi/0f0bf842-6353-49ed-91c0-c9d672f21200)?

Comment (moan0s) [reply docs/configuring-playbook-collabora.md:6]: Yes, I forgot to add that!

**Pull request merged**